legion/pummaMM/taco-generated-tblis.cpp

```diff
@@ -0,0 +1,773 @@
+  computePartitions.aPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.aPartition.valsPartition = a_vals_partition;
+  computePartitions.aPartition.denseLevelRunPartitions[0] = a_dense_run_0_Partition_0;
+  computePartitions.bPartition.indicesPartitions = std::vector<std::vector<Legion::LogicalPartition>>(2);
+  computePartitions.bPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.bPartition.valsPartition = b_vals_partition;
+  computePartitions.bPartition.denseLevelRunPartitions[0] = b_dense_run_0_Partition_0;
+  computePartitions.cPartition.indicesPartitions = std::vector<std::vector<Legion::LogicalPartition>>(2);
+  computePartitions.cPartition.denseLevelRunPartitions = std::vector<IndexPartition>(2);
+  computePartitions.cPartition.valsPartition = c_vals_partition;
+  computePartitions.cPartition.denseLevelRunPartitions[0] = c_dense_run_0_Partition_0;
+
+  return computePartitions;
+}
+
+void task_7(const Task* task, const std::vector<PhysicalRegion>& regions, Context ctx, Runtime* runtime) {
+  PhysicalRegion a_vals = regions[0];
+  LogicalRegion a_vals_parent = regions[0].get_logical_region();
+  PhysicalRegion b_vals = regions[1];
+  LogicalRegion b_vals_parent = regions[1].get_logical_region();
+  PhysicalRegion c_vals = regions[2];
+  LogicalRegion c_vals_parent = regions[2].get_logical_region();
+
+  int64_t iln = task->index_point[0];
+  task_7Args* args = (task_7Args*)(task->args);
+  Legion::FieldID a_vals_field_id = args->a_vals_field_id;
+  Legion::FieldID b_vals_field_id = args->b_vals_field_id;
+  Legion::FieldID c_vals_field_id = args->c_vals_field_id;
+
+  auto b_vals_ro_accessor = createAccessor<AccessorROdouble2>(b_vals, b_vals_field_id);
+  auto c_vals_ro_accessor = createAccessor<AccessorROdouble2>(c_vals, c_vals_field_id);
+  auto a_vals_rw_accessor = createAccessor<AccessorRWdouble2>(a_vals, a_vals_field_id);
+
+  auto aDomain = runtime->get_index_space_domain(ctx, a_vals.get_logical_region().get_index_space());
+  auto bDomain = runtime->get_index_space_domain(ctx, b_vals.get_logical_region().get_index_space());
+  auto cDomain = runtime->get_index_space_domain(ctx, c_vals.get_logical_region().get_index_space());
+  if (bDomain.get_volume() == 0 || cDomain.get_volume() == 0)
+    return ;
+
+  auto aShape = std::vector<tblis::len_type>(2);
+  aShape[0] = 1 + (aDomain.hi()[0] - aDomain.lo()[0]);
+  aShape[1] = 1 + (aDomain.hi()[1] - aDomain.lo()[1]);
+  auto bShape = std::vector<tblis::len_type>(2);
+  bShape[0] = 1 + (bDomain.hi()[0] - bDomain.lo()[0]);
+  bShape[1] = 1 + (bDomain.hi()[1] - bDomain.lo()[1]);
+  auto cShape = std::vector<tblis::len_type>(2);
+  cShape[0] = 1 + (cDomain.hi()[0] - cDomain.lo()[0]);
+  cShape[1] = 1 + (cDomain.hi()[1] - cDomain.lo()[1]);
+  auto aStrides = std::vector<tblis::stride_type>(2);
+  aStrides[0] = a_vals_rw_accessor.accessor.strides[0] / sizeof(double);
```

Comment:
Just double check that these stride values are what you and TBLIS expects. IIRC, for an N-dimensional region, `strides[i]` is the number of bytes between each entry of dimension `i`. I've had to manipulate this is other cases, so it might not be exactly what TBLIS is expecting.

Reply:
I think this is correct. TBLIS expects the stride to be number of underlying data type values, rather than number of bytes.